lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/api/VortexFuture.java

```diff
@@ -34,37 +36,89 @@
   // Otherwise tasklet has not completed.
   private Optional<TOutput> userResult = null;
   private Exception userException;
+  private boolean cancelled = false;
   private final CountDownLatch countDownLatch = new CountDownLatch(1);
   private final ThreadPoolStage<TOutput> stage;
+  private final VortexMaster vortexMaster;
+  private final int taskletId;
 
   /**
    * Creates a {@link VortexFuture}.
    */
-  public VortexFuture() {
+  @Private
+  public VortexFuture(final VortexMaster vortexMaster, final int taskletId) {
+    this.vortexMaster = vortexMaster;
+    this.taskletId = taskletId;
     stage = null;
   }
 
   /**
    * Creates a {@link VortexFuture} with a callback.
    */
-  public VortexFuture(final EventHandler<TOutput> callbackHandler) {
+  @Private
+  public VortexFuture(final VortexMaster vortexMaster,
```

Comment:
I'm a bit iffy about exposing ``VortexMaster`` and linking a taskletId with ``VortexFuture`` here. I was trying to instantiate with injection but it doesn't seem like we can inject something without knowing the type either. Any other solutions?